src/webapp/run_rebuild.sh

```diff
@@ -32,42 +32,78 @@ SOURCE=${BASH_SOURCE[0]}
 SCRIPT_DIR="$(dirname "$SOURCE")"
 cd "$SCRIPT_DIR" || exit 1
 
-# Need to check free space and limit Node memory usage
-# for PIs with little memory
-MemTotal=$(grep MemTotal /proc/meminfo | awk '{print $2}')
-MemFree=$(grep MemFree /proc/meminfo | awk '{print $2}')
-SwapFree=$(grep SwapFree /proc/meminfo | awk '{print $2}')
-TotalFree=$((SwapFree + MemFree))
-
-MemTotal=$((MemTotal / 1024))
-MemFree=$((MemFree / 1024))
-SwapFree=$((SwapFree / 1024))
-TotalFree=$((TotalFree / 1024))
-
-echo "Total phys memory: ${MemTotal} MB"
-echo "Free phys memory : ${MemFree} MB"
-echo "Free swap memory : ${SwapFree} MB"
-echo "Free total memory: ${TotalFree} MB"
-
-
-if [[ -z $NODEMEM ]]; then
-  # Keep a buffer of minimum 20 MB
-  if [[ $TotalFree -gt 1044 ]]; then
-    NODEMEM=1024
-  elif [[ $TotalFree -gt 532 ]]; then
-    NODEMEM=512
-  elif [[ $TotalFree -gt 276 ]]; then
-    NODEMEM=256
-  else
-    echo "ERROR: Not enough memory available on system. Please increase swap size to give at least 276 MByte free memory."
-    echo "Current free memory = $TotalFree MB"
-    echo "Hint: if only a little memory is missing, stopping spocon, mpd, and jukebox-daemon might give you enough space"
-    exit 1
+change_swap() {
+    local new_swap_size="$1"
+    sudo dphys-swapfile swapoff || return 1
+    sudo sed -i "s|.*CONF_SWAPSIZE=.*|CONF_SWAPSIZE=${new_swap_size}|g" /etc/dphys-swapfile || return 1
+    sudo sed -i "s|^\s*CONF_SWAPFACTOR=|#CONF_SWAPFACTOR=|g" /etc/dphys-swapfile || return 1
+    sudo dphys-swapfile setup || return 1
+    sudo dphys-swapfile swapon || return 1
+}
+
+calc_nodemem() {
+    # keep a buffer for the kernel etc.
+    local mem_buffer=256
+    # Need to check free space and limit Node memory usage
+    # for PIs with little memory
+    MemTotal=$(grep MemTotal /proc/meminfo | awk '{print $2}')
+    MemFree=$(grep MemFree /proc/meminfo | awk '{print $2}')
+    SwapTotal=$(grep SwapTotal /proc/meminfo | awk '{print $2}')
+    SwapFree=$(grep SwapFree /proc/meminfo | awk '{print $2}')
+    TotalFree=$((SwapFree + MemFree))
+
+    MemTotal=$((MemTotal / 1024))
+    MemFree=$((MemFree / 1024))
+    SwapTotal=$((SwapTotal / 1024))
+    SwapFree=$((SwapFree / 1024))
+    TotalFree=$((TotalFree / 1024))
+    FreeToUse=$((TotalFree - mem_buffer))
+
+    echo "Total phys memory : ${MemTotal} MB"
+    echo "Free phys memory  : ${MemFree} MB"
+    echo "Total swap memory : ${SwapTotal} MB"
+    echo "Free swap memory  : ${SwapFree} MB"
+    echo "Free total memory : ${TotalFree} MB"
+    echo "Keep as buffer    : ${mem_buffer} MB"
+    echo "Free usable memory: ${FreeToUse} MB"
```

Comment:
I know it has been there before, but we we need to show as many params to the user? I think it's fine just to show `${FreeToUse}`

Reply:
I added a "verbose" paramter and moved most logging information to it, in case some debugging is needed.